src/host/main.cc

```diff
@@ -1,52 +1,248 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <GLFW/glfw3.h>
+
 #include "burner.h"
 
 // Linux headers
 #include <dlfcn.h>
+#include <time.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <errno.h>
+#include <string.h>
 
 #define TITLE "burner"
 
+#if defined(DEVELOPMENT_MODE)
+    #warning "DEV"
+
+	#define call_start(game__, state__, renderer__, user_data__) (* game__.start)(state__, renderer__, user_data__)
+	#define call_update(game__, state__, renderer__, input__, time__, delta__, user_data__) (* game__.update)(state__, renderer__, input__, time__, delta__, user_data__)
+	#define call_on_reload(game__, state__, renderer__, user_data__) (* game__.on_reload)(state__, renderer__, user_data__)
+#else
+    #warning "RELEASE"
+
+	#define call_start(game__, state__, renderer__, user_data__) start(state__, renderer__, user_data__)
+	#define call_update(game__, state__, renderer__, input__, time__, delta__, user_data__) update(state__, renderer__, input__, time__, delta__, user_data__)
+	#define call_on_reload(game__, state__, renderer__, user_data__)
+
+#endif
```

Comment:
callbacks for the dynamic version, regular function calls for the static version (though I expect that you'd want to be able to keep live editing when exhibiting work anyway, so the static version would be for the absolute final build)